component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java

```diff
@@ -115,25 +115,37 @@ public BuilderImpl(final Schema providedSchema) {
             this.providedSchema = providedSchema;
         }
 
-        private void validateTypeAgainstProvidedSchema(final String name, final Schema.Type type, final Object value) {
-            if (providedSchema == null) {
-                return;
-            }
-            if (entryIndex == null) {
-                entryIndex = providedSchema.getEntries().stream().collect(toMap(Schema.Entry::getName, identity()));
+        private Schema.Entry findMandatoryEntry(final String name) {
```

Comment:
function name: remove mandatory or rename it ...

Reply:
renamed findExistingEntry, better ? why mandatory was not good ?

Reply:
sorry a matter of taste... ;-)